server/src/main/java/org/opensearch/action/admin/cluster/shards/TransportCatShardsAction.java

```diff
@@ -73,6 +85,10 @@ protected void innerOnFailure(Exception e) {
             client.admin().cluster().state(clusterStateRequest, new ActionListener<ClusterStateResponse>() {
                 @Override
                 public void onResponse(ClusterStateResponse clusterStateResponse) {
+                    if (shardsRequest.isRequestLimitCheckSupported()
+                        && requestLimitSettings.isCircuitLimitBreached(clusterStateResponse.getState(), CAT_SHARDS)) {
+                        listener.onFailure(new CircuitBreakingException("Too many shards requested.", CircuitBreaker.Durability.TRANSIENT));
```

Comment:
Should we also include the Limit in error message, so that user is aware of what is being supported?
Something like:
```"Too many shards requested. Can not request shards beyond {500}"```